File: crates/juicefront/ui/src/lib/more-menu.ts
import { t, type Locale } from "../i18n";

export interface MoreMenuItem {
  href: string;
  icon: string;
  label: string;
  jsOnly?: boolean;
}

export type MoreMenuEntry = MoreMenuItem | null;

export function getMoreMenuItems(
  locale: Locale,
  lp: (path: string) => string,
): MoreMenuEntry[] {
  return [
    { href: "#shortcuts-modal", icon: "command", label: t(locale, "nav.shortcuts"), jsOnly: true },
    { href: lp("/docs"), icon: "notebook", label: t(locale, "nav.docs") },
    { href: lp("/faq"), icon: "circle-help", label: t(locale, "nav.faq") },
    { href: "#share-modal", icon: "share", label: t(locale, "nav.share") },
    { href: "#pair-modal", icon: "smartphone", label: t(locale, "nav.pair_app"), jsOnly: true },
    { href: "#language-modal", icon: "globe", label: t(locale, "nav.language") },
    null,
    { href: lp("/feedback"), icon: "message-square", label: t(locale, "nav.feedback") },
    { href: lp("/terms"), icon: "shield", label: t(locale, "nav.terms") },
    { href: lp("/privacy"), icon: "lock", label: t(locale, "nav.privacy") },
  ];
}

File: crates/juicefront/src/ssr/more_menu.rs
use serde::Serialize;

use crate::i18n;

#[derive(Debug, Clone, Serialize)]
pub struct MenuItem {
    pub href: String,
    pub icon: String,
    pub label: String,
    pub js_only: bool,
}

#[derive(Debug, Clone, Serialize)]
pub struct MenuEntry {
    pub item: Option<MenuItem>,
}

pub fn more_menu_items(locale: &str) -> Vec<MenuEntry> {
    let entry = |href: String, icon: &str, key: &str, js_only: bool| MenuEntry {
        item: Some(MenuItem {
            href,
            icon: icon.to_owned(),
            label: i18n::t(locale, key),
            js_only,
        }),
    };
    vec![
        entry(
            "#shortcuts-modal".to_owned(),
            "command",
            "nav.shortcuts",
            true,
        ),
        entry(i18n::lp(locale, "/docs"), "notebook", "nav.docs", false),
        entry(i18n::lp(locale, "/faq"), "circle-help", "nav.faq", false),
        entry("#share-modal".to_owned(), "share", "nav.share", false),
        entry("#pair-modal".to_owned(), "smartphone", "nav.pair_app", true),
        entry("#language-modal".to_owned(), "globe", "nav.language", false),
        MenuEntry { item: None },
        entry(
            i18n::lp(locale, "/feedback"),
            "message-square",
            "nav.feedback",
            false,
        ),
        entry(i18n::lp(locale, "/terms"), "shield", "nav.terms", false),
        entry(i18n::lp(locale, "/privacy"), "lock", "nav.privacy", false),
    ]
}
